src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/CsvTestData.kt

```diff
@@ -267,6 +268,247 @@ object CsvTestData {
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "HLI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "IWI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "LTI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "SFI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "UKI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "WDI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "WTI"),
+      OfferingRecord(course = "Healthy Sex Programme", prisonId = "WMI"),
+      OfferingRecord(course = "Horizon", prisonId = "ASI"),
+      OfferingRecord(course = "Horizon", prisonId = "BNI"),
+      OfferingRecord(course = "Horizon", prisonId = "BRI"),
+      OfferingRecord(course = "Horizon", prisonId = "CWI"),
+      OfferingRecord(course = "Horizon", prisonId = "DRI"),
+      OfferingRecord(course = "Horizon", prisonId = "FKI"),
+      OfferingRecord(course = "Horizon", prisonId = "GHI"),
+      OfferingRecord(course = "Horizon", prisonId = "HLI"),
+      OfferingRecord(course = "Horizon", prisonId = "IWI"),
+      OfferingRecord(course = "Horizon", prisonId = "LTI"),
+      OfferingRecord(course = "Horizon", prisonId = "MDI"),
+      OfferingRecord(course = "Horizon", prisonId = "NLI"),
+      OfferingRecord(course = "Horizon", prisonId = "RHI"),
+      OfferingRecord(course = "Horizon", prisonId = "SFI"),
+      OfferingRecord(course = "Horizon", prisonId = "SNI"),
+      OfferingRecord(course = "Horizon", prisonId = "UKI"),
+      OfferingRecord(course = "Horizon", prisonId = "WTI"),
+      OfferingRecord(course = "Horizon", prisonId = "WHI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "AYI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "BAI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "DNI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "DGI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "FYI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "FKI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "FNI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "GHI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "GTI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "ISI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "LLI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "MRI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "SLI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "SNI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "MTI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "WRI"),
+      OfferingRecord(course = "Identity Matters", prisonId = "WMI"),
+      OfferingRecord(course = "Kaizen (sexual offences)", prisonId = "ASI"),
```

Comment:
I wonder if we need to do this by some sort of `id` here - the Course name will just be `Kaizen` but the `(sexual offences)` will be the audience so we'll need a way of distinguishing between two courses with the same `name` but different `audience`s... Alternatively we could use `course` and `audience(s)` as a composite foreign key here. 

I've realised this also applies to prerequisites so we'll need to apply the same approach there. Any thoughts on what might be easier? Happy to help update the CSV for this.

Reply:
Seems to me that these are different courses that happen to have the same name. 
If HMPPS want to give the same name to different courses they're probably going to regret it.
We can't use audience(s) as part of a composite key because audience is separate entity in a many-to-many relationship with course. You could could either use different course names - Kaizen for sexual offences.

Reply:
They're the same courses, but different "strands" that we're representing with the `audience` fields (see designs [here](https://www.figma.com/proto/ZGMNCyn4uzmUxFQdmAiQQq/Accredited-Programmes-Prototype?type=design&node-id=1309-8923&scaling=min-zoom&page-id=892%3A7020&starting-point-node-id=1309%3A9318)) for how they're displayed.

Could you elaborate on the following? Could we not use the list of audience IDs and the course's name
 to identify it here?
> We can't use audience(s) as part of a composite key because audience is separate entity in a many-to-many relationship with course.